--- /prog/pwm/pwmconfig.orig
+++ /prog/pwm/pwmconfig
@@ -157,6 +157,14 @@
 function is_pwm_auto()
 {
 	local ENABLE=${1}_enable
+	local NAME="$(dirname ${1})/name"
+	local DRVNAME="$(cat $NAME)"
+
+	# pwmfan driver does not implement auto mode
+	if [ "$DRVNAME" = "pwmfan" ]
+	then
+		return 1
+	fi
 
 	if [ -f $ENABLE ]
 	then
@@ -173,9 +181,12 @@
 function pwmdisable()
 {
 	local ENABLE=${1}_enable
+	local NAME="$(dirname ${1})/name"
+	local DRVNAME="$(cat $NAME)"
 
-	# No enable file? Just set to max
-	if [ ! -f $ENABLE ]
+	# No enable file? Just set to max;
+	# pwmfan driver implements enable file, but with different purpose
+	if [ ! -f $ENABLE ] || [ "$DRVNAME" = "pwmfan" ]
 	then
 		echo $MAX > $1
 		return 0
@@ -213,8 +224,12 @@
 function pwmenable()
 {
 	local ENABLE=${1}_enable
+	local NAME="$(dirname ${1})/name"
+	local DRVNAME="$(cat $NAME)"
 
-	if [ -w $ENABLE ]
+	# enable pwm, but only if driver is not pwmfan since
+	# this one implements enable with different purpose
+	if [ -w $ENABLE ] && [ "$DRVNAME" != "pwmfan" ]
 	then
 		echo 1 2>/dev/null > $ENABLE
 		if [ $? -ne 0 ]
